src/binding.cc

```diff
@@ -73,119 +33,117 @@ napi_status napix_get_instance_data(
       });
-    }
 
-    // If AB_CREATED_BY_REF, then another finalizer has been added before this
-    // as a "real" backing store finalizer.
-    if (mode != AB_CREATED_BY_REF) {
-      buf.AddFinalizer([this](Env env, char* ptr) {
-        UnregisterArrayBuffer(ptr);
-      }, ptr);
-    }
-  }
+  exports.Set("PointerBuffer", func);
+  InstanceData *data = InstanceData::Get(env);
+  data->pointer_ctor = Persistent(func);
+  return exports;
+}
 
-  inline void UnregisterArrayBuffer(char* ptr) {
-    auto it = pointer_to_orig_buffer.find(ptr);
-    if (--it->second.finalizer_count == 0)
-      pointer_to_orig_buffer.erase(it);
+PointerBuffer::PointerBuffer(Napi::CallbackInfo& info) : Napi::ObjectWrap<PointerBuffer>(info) {
+  Napi::Env env = info.Env();
+  int length = info.Length();
+  if (length <= 0 || !info[0].IsNumber()) {
+    Napi::TypeError::New(env, "Number expected").ThrowAsJavaScriptException();
+    return;
   }
+  Napi::Number value = info[0].As<Napi::Number>();
+  ptr_ = reinterpret_cast<char *>(value.Int64Value());
+  length_ = (ptr_ == nullptr ? 0 : info[1].As<Number>().Int32Value());
+}
 
-  inline ArrayBuffer LookupOrCreateArrayBuffer(char* ptr, size_t length) {
-    assert(ptr != nullptr);
-    ArrayBuffer ab;
-    auto it = pointer_to_orig_buffer.find(ptr);
-    if (it != pointer_to_orig_buffer.end())
-      ab = it->second.ab.Value();
-
-    if (ab.IsEmpty()) {
-      length = std::max<size_t>(length, kMaxLength);
-      ab = Buffer<char>::New(env, ptr, length, [this](Env env, char* ptr) {
-        UnregisterArrayBuffer(ptr);
-      }).ArrayBuffer();
-      RegisterArrayBuffer(ab, AB_CREATED_BY_REF);
-    }
-    return ab;
-  }
+Napi::Value PointerBuffer::IsNull(const Napi::CallbackInfo &info) {
+  return Boolean::New(info.Env(), ptr_ == nullptr);
+}
 
-  napi_value WrapPointer(char* ptr, size_t length) override;
-  char* GetBufferData(napi_value val) override;
+Napi::Value PointerBuffer::Address(const Napi::CallbackInfo &info) {
+  return Number::New(info.Env(), reinterpret_cast<int64_t>(ptr_));
+}
 
-  static InstanceData* Get(Env env) {
-    void* d = nullptr;
-    if (napix_get_instance_data(env, &d) == napi_ok)
-      return static_cast<InstanceData*>(d);
-    return nullptr;
+Napi::Value PointerBuffer::Length(const Napi::CallbackInfo &info) {
+  return Number::New(info.Env(), length_);
+}
+
+Napi::Value PointerBuffer::Get(const Napi::CallbackInfo &info) {
+  int32_t offset = info[0].As<Number>().Int32Value();
+  return Number::New(info.Env(), ptr_[offset]);
+}
+
+
+Napi::Value PointerBuffer::ToString(const Napi::CallbackInfo &info) {
+  Napi::Env env = info.Env();
+  int length = info.Length();
+  if (length == 1 || info[0].IsString()) {
+     std::string encoding = info[0].As<String>();
+     if (encoding == "utf-8") {
+        return String::New(info.Env(), ptr_, length_);
+     } else if (encoding == "ucs2") {
+        return String::New(info.Env(), reinterpret_cast<char16_t*>(ptr_), length_ / 2);
```

Comment:
I’d probably just provide a way to read a slice of N bytes starting this pointer as a Buffer (but as a copy, not backed by this address), and then use `.toString()` on that, so that we keep being compatible with existing code